src/github.com/matrix-org/dendrite/federationapi/routing/state.go

```diff
@@ -68,6 +78,26 @@ func GetStateIDs(
 	}
 }
 
+func parseEventIDParam(
+	request *gomatrixserverlib.FederationRequest,
+) (eventID string, resErr *util.JSONResponse) {
+	URL, err := url.Parse(request.RequestURI())
```

Comment:
Was there a reason you parsed the URL first here instead of simply doing:

```
eventID = req.URL.Query().Get("event_id")
```

Like we do elsewhere? https://github.com/matrix-org/dendrite/blob/b86b55360a6b46b5edc0f6d049fd997f46a9b074/src/github.com/matrix-org/dendrite/clientapi/auth/auth.go#L159

Reply:
Intention of `request *gomatrixserverlib.FederationRequest` is to be a signature verified version of raw federation `http.Request`.
https://github.com/matrix-org/gomatrixserverlib/blob/4c45af876608399b13cf891e2d9f89f01c67137a/request.go#L183-L194

Although it may be argued that after verification, we are allowed to use `http.Request`'s parameters, it's safer to just use `request *gomatrixserverlib.FederationRequest`.

Reply:
Fair enough.